### PR TITLE
FLUME-3093: Groundwork for version changes in root pom

### PR DESCRIPTION
**Pull request by mcsanady**
Extracted hard-coded version numbers to root pom.xml

No version number changed in this step. The old used versions are named with -old suffix.
 guava-old.version
 httpclient-old.version
 mvn-surefire-plugin-old.version
 mvn-surefire-plugin-old-morphline.version

### CODE REVIEW

**Comment by mcsanady** [comment]
Tested with mvn dependency:tree : no difference detected.

**Review by mcsanady — COMMENTED** [review]
Removed unnecessary flume.version variable.
Testing: mvn dependency:tree outputs are identical

**Comment by mcsanady** [comment]
I accidentally deleted the branch

File: flume-ng-auth/pom.xml [hunk 1]
```diff
@@ -37,7 +37,7 @@ limitations under the License.
       <plugin>
         <groupId>org.apache.felix</groupId>
         <artifactId>maven-bundle-plugin</artifactId>
-        <version>2.3.7</version>
+        <version>${bundle-plugin.version}</version>
```
Comment (szaboferee) [inline flume-ng-auth/pom.xml]: why don't we use `<pluginManagement>` in the parent pom?

File: flume-ng-channels/flume-spillable-memory-channel/pom.xml [hunk 2]
```diff
@@ -86,14 +86,14 @@
     <dependency>
       <groupId>org.mockito</groupId>
       <artifactId>mockito-all</artifactId>
-      <version>1.9.0</version>
+      <version>${mockito.version}</version>
```
Comment (szaboferee) [inline flume-ng-channels/flume-spillable-memory-channel/pom.xml]: Why don't we use `<dependencyManagement>` in the parent pom

File: flume-ng-node/pom.xml [hunk 3]
```diff
@@ -147,7 +147,7 @@
     <dependency>
       <groupId>com.github.stefanbirkner</groupId>
       <artifactId>system-rules</artifactId>
-      <version>1.16.0</version>
+      <version>${system-rules.version}</version>
```
Comment (szaboferee) [inline flume-ng-node/pom.xml]: dependencyManagement?

File: flume-ng-sinks/flume-http-sink/pom.xml [hunk 4]
```diff
@@ -30,11 +30,6 @@ limitations under the License.
   <artifactId>flume-http-sink</artifactId>
   <name>Flume HTTP/S Sink</name>
 
-  <properties>
-    <wiremock.version>1.53</wiremock.version>
-    <guava.version>18.0</guava.version>
-  </properties>
-
```
Comment (szaboferee) [inline flume-ng-sinks/flume-http-sink/pom.xml:37]: Now, that we use common versions, we might would like to look at the exlusions in the dependencies. Again, the version numbers could come from `dependencyManagement`

**Comment by simonati** [comment]
Hi @mcsanady, 
I think this is a neat idea I would like to finish my review soon.

**Pull request closed without merging**